components/builder-api/src/depot/server.rs

```diff
@@ -2425,8 +2418,18 @@ fn process_upload_for_package_archive(
 
     // Don't re-create the origin package if it already exists
     if !origin_package_found {
-        if let Err(err) = conn.route::<OriginPackageCreate, OriginPackage>(&package) {
-            return Err(err);
+        match conn.route::<OriginPackageCreate, OriginPackage>(&package) {
+            Ok(pkg) => {
+                let mut job_graph_package = JobGraphPackageCreate::new();
+                job_graph_package.set_package(pkg);
+
+                if let Err(err) =
+                    conn.route::<JobGraphPackageCreate, OriginPackage>(&job_graph_package)
+                {
+                    return Err(err);
```

Comment:
If this particular branch is reached, it means that the `OriginPackageCreate` succeeded, but the `JobGraphPackageCreate` failed.  So effectively, we've created a package that doesn't exist in our package graph.  Handling that kind of error gracefully feels pretty tricky, and likely something we can punt on for this particular PR (since it will become irrelevant as the services are consolidated), but at the very least, we should log an error with `warn!`, so if this ends up biting us, we can track it down easier.